### PR TITLE
fix(kanban): preserve order by name

### PR DESCRIPTION
**Pull request by stonebuzz**
preserve column order by name -and handle translations)

Before : 
image

After : 
image

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !28659

### CODE REVIEW

File: src/Project.php [hunk 1]
```diff
@@ -1962,10 +1962,11 @@ public static function getAllKanbanColumns($column_field = null, $column_ids = [
             if (!empty($column_ids) && !$get_default) {
                 $restrict = ['id' => $column_ids];
             }
-            $allstates = $projectstate->find($restrict, ['is_finished ASC', 'id']);
+            $allstates = $projectstate->find($restrict, ['name ASC']);
```
Comment (cedric-anne) [inline src/Project.php:1965]: If you want to sot by name, you should probably do it in PHP to be able to use a natural sorting method.

Also, I noticed that dropdown translations are not used here. Maybe they should.

Comment (stonebuzz) [reply src/Project.php:1965]: Last commit do a sort from PHP and handle translation

File: src/Project.php [hunk 2]
```diff
@@ -34,6 +34,7 @@
  */
 
 use Glpi\Application\View\TemplateRenderer;
+use Glpi\Debug\Toolbar;
```
Comment (cedric-anne) [inline src/Project.php:37]:
```suggestion
```

**Comment by cconard96** [comment]
The original intent was to have the statuses shown with the statuses marked as "Is finished" at the bottom. It also showed them in order based on the lifecycle if you only have the default statuses. I'm not sure how having them alphabetical would affect UX, but I'm not against the change.

**Review by cconard96 — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

File: src/Project.php [hunk 3]
```diff
@@ -1956,25 +1956,64 @@ public static function getAllForKanban($active = true, $current_id = -1)
...
+                        'name'            => $projectstate['transname'] ?? $projectstate['name'],
+                        'id'              => $projectstate['id'],
+                        'header_color'    => $projectstate['color'],
+                        'header_fg_color' => Toolbox::getFgColor($projectstate['color'], 50),
+                    ];
+                }
+            }
+
+            // sort by name ASC
+            usort($result, function($a, $b) {
```
Comment (cedric-anne) [inline src/Project.php:2011]:
```suggestion
            usort($result, function ($a, $b) {
```

**Pull request merged**